sdk-cpp/src/do_download.cpp

```diff
@@ -22,39 +21,44 @@ namespace deliveryoptimization
 
 download::download(const std::string& uri, const std::string& downloadFilePath)
 {
-    _download = std::make_shared<msdod::CDownloadImpl>(uri, downloadFilePath);
+    _download = std::make_shared<msdod::CDownloadImpl>();
+    _download->Init(uri, downloadFilePath);
```

Comment:
>Init [](http://example.com/codeflow?start=15&length=4)

Ignores errors from the Init method. Do you intend to handle it in a separate PR?